### PR TITLE
adding Shared responsibility model overview

### PR DESCRIPTION
**Pull request by SharpRake**
## Type of change

This doc adds an overview of Chainguard's Shared Responsibility model, which outlines who is responsible for what aspects of the security of Chainguard Images. 

### What should this PR do?

Resolves https://github.com/chainguard-dev/internal/issues/3911

### Why are we making this change?

Support customers by helping them understand what aspects of container security they're responsible for (in the context of Chainguard Images)

### What are the acceptance criteria? 

Would love @MordodeMaru's :+1: or :-1: 

Note that one big change from [the original doc/request](https://docs.google.com/document/d/1DvJdr-6KNqyVuDAAe8Pmc6hrFNi42NqKwro17lIVnqU/edit?usp=sharing) is the table. The original request included an image of the table but I found it too hard to read on Academy. I translated it to a markdown table but to help make it less busy I cut some of the content. LMK if this needs to be looked at further.

### How should this PR be tested?

Check for content and formatting. No tech test to speak of.

### CODE REVIEW

File: content/chainguard/chainguard-images/shared-responsibility-model/index.md [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+## App Images
+Chainguard App Images are specialized container images designed to be used "as-is" in production environments.
+
+### Key Characteristics
+1. Minimal Attack Surface: Built using a [distroless](/chainguard/chainguard-images/getting-started-distroless/) approach, removing unnecessary components to reduce potential vulnerabilities.
+2. Up-to-date Components: Continuously updated with the latest security patches and version updates from upstream projects.
+3. Low to Zero CVEs: Rigorously scanned and maintained to minimize known vulnerabilities.
+4. Layered Security Responsibility:
+    * Top layers (Apps and Lang Libs) are managed by upstream open source projects
+    * Lower layers (Toolchains and System) are secured and maintained by Chainguard
```
Comment (amouat) [inline content/chainguard/chainguard-images/shared-responsibility-model/index.md:41]: I guess this wording comes from elsewhere, but I wonder if it's misleading. I mean the toolchains and system layers are themselves open source projects...

File: content/chainguard/chainguard-images/shared-responsibility-model/index.md [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+5. Compliance with CHP Levels: Adhere to Chainguard's Container Hardening Policy, typically at higher levels (CHP3-CHP5) for enhanced security.
```
Comment (amouat) [inline content/chainguard/chainguard-images/shared-responsibility-model/index.md:42]: I thought CHPs this was just a proposal! Mention that they are explained below, maybe just link?

File: content/chainguard/chainguard-images/shared-responsibility-model/index.md [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+### Use Cases
+* Ideal for organizations requiring secure, production-ready container images
+* Suitable for environments with stringent security and compliance requirements
+* Designed to integrate seamlessly into modern CI/CD pipelines and cloud-native architectures
+
+### App Images Shared Responsibilities
+* Apps Layer: Management of CVEs in the top-level application code.
+* Lang Libs Layer: Handling vulnerabilities in language-specific libraries built into the image (e.g., `go mod`, `pom`).
+
+Should a CVE be detected in any of these images we would likely bump a dependency to avoid in the next build. The outcome is two-fold: either bumping the dependency is successful or it breaks break the image.
```
Comment (amouat) [inline content/chainguard/chainguard-images/shared-responsibility-model/index.md:58]: First sentence needs rewritten. Maybe:

"If a CVE is detected in one of these layers and is fixable by updating a dependency, we will attempt to do so."?

In general I don't like the term "bumping" for customer facing docs, but that could just be me :)

Comment (amouat) [reply content/chainguard/chainguard-images/shared-responsibility-model/index.md:58]: I guess the second sentence should be: "There are two possible outcomes; ..."

To me "two-fold" implies both things are true, not a binary decision.

File: content/chainguard/chainguard-images/shared-responsibility-model/index.md [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+* Suitable for environments with stringent security and compliance requirements
+* Designed to integrate seamlessly into modern CI/CD pipelines and cloud-native architectures
+
+### App Images Shared Responsibilities
+* Apps Layer: Management of CVEs in the top-level application code.
+* Lang Libs Layer: Handling vulnerabilities in language-specific libraries built into the image (e.g., `go mod`, `pom`).
+
+Should a CVE be detected in any of these images we would likely bump a dependency to avoid in the next build. The outcome is two-fold: either bumping the dependency is successful or it breaks break the image. 
+
+* **Bumping a dependency is successful**: In this case, the upgrade does not cause the application to break,\ and the CVE is considered independently fixable. This falls under the "Release or Rebuild" condition for a qualifying patch in our SLAs, where the remediation involves rebuilding the image with updated compilers or libraries. The CVE clock starts, and Chainguard aims to resolve the vulnerability within the SLA's time frame.
```
Comment (amouat) [inline content/chainguard/chainguard-images/shared-responsibility-model/index.md:60]: rogue backslash

File: content/chainguard/chainguard-images/shared-responsibility-model/index.md [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+* **Bumping a dependency is breaking**: If the upgrade causes the application to break, the CVE is not considered independently fixable. Chainguard will file a "pending-upstream-fix" advisory, indicating that the vulnerability cannot be fixed without a fix from the upstream project. Chainguard cannot always service vulnerabilities from the "Lang Lib" (language libraries) layer, unfortunately. The CVE clock does not start in this case, as the vulnerability is not considered fixable under the SLA.
+
+> **Note**: To learn more about Chainguard's Security Advisories, please refer to our [overview on the subject](/chainguard/chainguard-images/working-with-images/security-advisories/). We also encourage you to check out the [Security Advisories page](https://images.chainguard.dev/security) directly. 
+
+### Chainguard's Responsibilities in App Images
+* Toolchains Layer: CVE removal in development tools and runtimes (e.g., `go`, `java`, `php`).
+* System Layer: Addressing vulnerabilities in base system libraries and utilities (e.g., `glibc`, `openssl`).
+
+### Customer's Responsibilities in App Images
+* Any added libraries or APK packages are the customer's responsibility.
```
Comment (amouat) [inline content/chainguard/chainguard-images/shared-responsibility-model/index.md:70]: Just to be clear here, we are talking about wolfi packages here? Or apks that the customer has created?

File: content/chainguard/chainguard-images/shared-responsibility-model/index.md [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+
+The next few sections dig deeper into the details about App Images, Base Images, and the responsibilities of Chainguard and its customers for both.
+
+
+## App Images
+Chainguard App Images are specialized container images designed to be used "as-is" in production environments.
+
+### Key Characteristics
+1. Minimal Attack Surface: Built using a [distroless](/chainguard/chainguard-images/getting-started-distroless/) approach, removing unnecessary components to reduce potential vulnerabilities.
```
Comment (amouat) [inline content/chainguard/chainguard-images/shared-responsibility-model/index.md:36]: could put the lead text e.g. "Minimal Attack Surface" in bold or similar? Same with the list below.

File: content/chainguard/chainguard-images/shared-responsibility-model/index.md [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+|  **CHP2** |      Read-write local filesystem     |                      Shell access                     |                              System level utilities                             |                             🔒<br>limited networking                             |      🔒<br>low to 0 CVEs     |
+|  **CHP1** |      Read-write local filesystem     |                      Shell access                     |                              System level utilities                             |                 Outbound network (apk, git, wget, curl) allowed                 |      🔒<br>low to 0 CVEs     |
+|  **CHP0** |      Read-write local filesystem     |                      Shell access                     |                              System level utilities                             |                 Outbound network (apk, git, wget, curl) allowed                 |      accumulating CVEs      |
+
+Chainguard Images that meet CHP levels 0 or 1 can be used as base images to build on top of. Images meeting CHP levels 2, 3, and 4 are to be used as application-specific images. Images that align with the requirements for CHP level 5 are truly distroless application-specific images.
```
Comment (amouat) [inline content/chainguard/chainguard-images/shared-responsibility-model/index.md:128]: > Chainguard Images that meet CHP levels 0 or 1 can be used as base images to build on top of.

This implies the others can't, so I think it should be reworded.

Maybe: "CHP levels 0 and 1 are typically found in base images used in intermediate build steps."?

Comment (amouat) [reply content/chainguard/chainguard-images/shared-responsibility-model/index.md:128]: >Images meeting CHP levels 2, 3, and 4 are to be used as application-specific images. Images that align with the requirements for CHP level 5 are truly distroless application-specific images.

This is a bit confusing. "are to be used"?

Comment (amouat) [reply content/chainguard/chainguard-images/shared-responsibility-model/index.md:128]: Also what does "accumulating CVEs" mean? An image that intentionally isn't being updated? I think that could be reworded, maybe "(not) continuously updated"?

**Review by amouat — CHANGES_REQUESTED** [review]
This seems to be a really important article! I think we should make sure everyone appropriate has signed off as there are policy decisions in here. Apologies if this has already happened, I just want to be cautious here.

The CHPs levels are a bit buried, I wasn't expecting those until they appeared.

I flagged a few other things, most minor, but a few things about CHPs I would like to get explained/clarified before publishing.

File: content/chainguard/chainguard-images/shared-responsibility-model/index.md [hunk 8]
```diff
@@ -0,0 +1,167 @@
...
+menu:
+   docs:
+    parent: "chainguard-images"
+weight: 575
+toc: true
+---
+
+Chainguard's Shared Responsibility Model is a comprehensive framework that delineates security responsibilities for hardened container images among Chainguard, upstream open-source projects, and customers. This model provides a clear understanding of who is responsible for what, ensuring a secure and compliant container environment. It addresses two primary categories: App Images (used as-is) and Base Images (used in multistage builds).
+
+This reference guide outlines the Shared Responsibility Model, and includes a helpful diagram outlining the relevant responsiblities. It is intended to serve as a resource for customers who want to better understand their security responsibilities (as well as Chainguard's responsibilities) in regards to Chainguard Images.
```
[the remark this answers is not in the document]
Comment (SharpRake) [reply content/chainguard/chainguard-images/shared-responsibility-model/index.md:22]: I tried cutting out the fluff from the second paragraph.

File: content/chainguard/chainguard-images/shared-responsibility-model/index.md [hunk 9]
```diff
@@ -0,0 +1,167 @@
...
+
+
+## Hardened Container Image Shared Responsibility Model (Diagram)
+
+![Diagram representing Chainguard's shared responsibility model. On the left is a pyramid structure labeled "App Images" with 4 tiers: Apps, LangLibs (go mod, pom, …), Toolchains (go, java, php, …), and System (glibc, openssl). To the left of the pyramid there are two brackets showing that CVE management is performed by the Upstream Open Source project at the Apps level and by Chainguard at the Toolchains and System level. To the right of the pyramid are two smaller brackets showing that if bumping a dependency is breaking, the CVE management falls to the Upstream project; if it's successful, then CVE management falls to Chainguard. On the right is another pyramid labeled "Base images" with 3 tiers: Custom code, Toolchains (go, java, php, …), and System (glibc, openssl, …). To the left of this pyramid are two brackets, showing that CVE management is the responsibility of the customer a the Custom code level and it's the responsibility of Chainguard at the Toolchains and System levels.](shared-responsibility-model.png)
```
[the remark this answers is not in the document]
Comment (SharpRake) [reply content/chainguard/chainguard-images/shared-responsibility-model/index.md:27]: Jordi is working on updating the Image. I've changed "bumping" ==> "updating" throughout the article

File: content/chainguard/chainguard-images/shared-responsibility-model/index.md (hunk 5 above)
Comment (SharpRake) [reply content/chainguard/chainguard-images/shared-responsibility-model/index.md:70]: I'm not sure, @MordodeMaru do you have insight?

File: content/chainguard/chainguard-images/shared-responsibility-model/index.md (hunk 1 above)
Comment (SharpRake) [reply content/chainguard/chainguard-images/shared-responsibility-model/index.md:41]: @MordodeMaru Could also use clarification here

**Comment by MordodeMaru** [comment]
> This seems to be a really important article! I think we should make sure everyone appropriate has signed off as there are policy decisions in here. Apologies if this has already happened, I just want to be cautious here.
> 
> The CHPs levels are a bit buried, I wasn't expecting those until they appeared.
> 
> I flagged a few other things, most minor, but a few things about CHPs I would like to get explained/clarified before publishing.

I agree with this and I don't think we've got alignment to publish this just yet. @SharpRake please put it on hold or, if you wish, close the PR for now. This is weeks away from being solid now.

**Comment by SharpRake** [comment]
Closing this PR for now, will draft a new one when we've aligned.

Thanks for the discussion all!

**Pull request closed without merging**